File: app/src/main/java/com/pmberjaya/indotiki/app/event/EventTab.java
package com.pmberjaya.indotiki.app.event;

import android.content.DialogInterface;
import android.content.Intent;
import android.os.Bundle;
import android.support.design.widget.AppBarLayout;
import android.support.design.widget.TabLayout;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;
import android.support.v4.view.ViewPager;
import android.support.v7.app.ActionBar;
import android.support.v7.app.AlertDialog;
import android.support.v7.widget.Toolbar;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;
import android.widget.LinearLayout;

import com.pmberjaya.indotiki.R;
import com.pmberjaya.indotiki.base.BaseActivity;
import com.pmberjaya.indotiki.dao.SessionManager;
import com.pmberjaya.indotiki.services.fcm.FCMAuthLoginHandler;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by edwin on 11/10/2017.
 */

public class EventTab extends BaseActivity implements ViewPager.OnPageChangeListener{

    private Toolbar toolbar;
    private TabLayout tabLayout;
    private ViewPager viewPager;
    private MenuItem item;
    private LinearLayout loading;
    SessionManager session;
    private AppBarLayout appBarLayout;
    private ViewPagerAdapter adapter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.event_tab);
        session = new SessionManager(EventTab.this);
        loading = (LinearLayout)findViewById(R.id.layout_loading);
        appBarLayout = (AppBarLayout)findViewById(R.id.app_bar);
        appBarLayout.setVisibility(View.VISIBLE);
        loading.setVisibility(View.GONE);
        initToolbar();
        viewPager = (ViewPager) findViewById(R.id.viewpager);
        setupViewPager(viewPager);
        tabLayout = (TabLayout) findViewById(R.id.tabs);
        tabLayout.setupWithViewPager(viewPager);
    }

    private void initToolbar(){
        toolbar = (Toolbar) findViewById(R.id.tool_bar);
        if(toolbar!=null) {
            setSupportActionBar(toolbar);
            toolbar.setNavigationOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
//                    onBackPressed();
                }
            });
        }
        ActionBar actionBar = getSupportActionBar();
        if(actionBar!=null) {
//            actionBar.setDisplayHomeAsUpEnabled(true);
            getSupportActionBar().setTitle(getResources().getString(R.string.news));
        }
    }

    private void setupViewPager(ViewPager viewPager) {
        adapter = new ViewPagerAdapter(getSupportFragmentManager());
        adapter.addFragment(new EventListPromoFragment(), getResources().getString(R.string.promo));
        adapter.addFragment(new EventListNewFragment(),getResources().getString(R.string.event_new));
        adapter.addFragment(new EventListOtherFragment(),getResources().getString(R.string.event_other));
        viewPager.setAdapter(adapter);
        viewPager.setOffscreenPageLimit(adapter.getCount());
    }

    @Override
    public void onPageScrolled(int position, float positionOffset, int positionOffsetPixels) {
        if(position == 0){
            item.setVisible(false);
        }
    }
    @Override
    public void onPageSelected(int position) {
        if(position == 0){
            item.setVisible(false);
        }
    }

    @Override
    public void onPageScrollStateChanged(int state) {

    }

    class ViewPagerAdapter extends FragmentPagerAdapter {
        private final List<Fragment> mFragmentList = new ArrayList<>();
        private final List<String> mFragmentTitleList = new ArrayList<>();

        public ViewPagerAdapter(FragmentManager manager) {
            super(manager);
        }

        @Override
        public Fragment getItem(int position) {
            return mFragmentList.get(position);
        }

        @Override
        public int getCount() {
            return mFragmentList.size();
        }

        public void addFragment(Fragment fragment, String title) {
            mFragmentList.add(fragment);
            mFragmentTitleList.add(title);
        }

        @Override
        public CharSequence getPageTitle(int position) {
            return mFragmentTitleList.get(position);
        }
    }

    @Override
    public boolean onPrepareOptionsMenu(Menu menu) {
        item = menu.findItem(R.id.action_filter);
        return super.onPrepareOptionsMenu(menu);
    }
    @Override
    public void onResume() {
        super.onResume();
    }
    @Override
    public void onPause() {
        super.onPause();
    }
    @Override
    public void onBackPressed()
    {
        AlertDialog dialog;
        AlertDialog.Builder builder = new AlertDialog.Builder(EventTab.this);
        builder.setTitle(null);
        builder.setMessage(R.string.exit_confirmation);
        builder.setIcon(R.mipmap.ic_launcher);
        // Set the action buttons
        builder.setPositiveButton(R.string.yes, new DialogInterface.OnClickListener() {
            @Override
            public void onClick(DialogInterface dialog, int id) {
//                  stopService(new Intent(MainMenuActivity.this, FCMRealtimeDatabaseHandler.class));
                stopService(new Intent(EventTab.this, FCMAuthLoginHandler.class));
                EventTab.this.finish();
            }
        });
        builder.setNegativeButton(R.string.cancel, new DialogInterface.OnClickListener() {
            @Override
            public void onClick(DialogInterface dialog, int id) {
                dialog.dismiss();
            }
        });
        dialog = builder.create();
        dialog.show();
        dialog.setCanceledOnTouchOutside(false);
    }
}

File: app/src/main/java/com/pmberjaya/indotiki/app/main/EventTab.kt
package com.pmberjaya.indotiki.app.main

import android.app.Activity
import android.os.Bundle
import android.support.design.widget.AppBarLayout
import android.support.design.widget.TabLayout
import android.support.v4.app.Fragment
import android.support.v4.app.FragmentActivity
import android.support.v4.app.FragmentManager
import android.support.v4.app.FragmentPagerAdapter
import android.support.v4.view.ViewPager
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.Toolbar
import android.view.LayoutInflater
import android.view.MenuItem
import android.view.View
import android.view.ViewGroup
import android.widget.LinearLayout
import com.pmberjaya.indotiki.R
import com.pmberjaya.indotiki.app.event.EventListNewFragment
import com.pmberjaya.indotiki.app.event.EventListOtherFragment
import com.pmberjaya.indotiki.app.event.EventListPromoFragment
import com.pmberjaya.indotiki.dao.SessionManager
import java.util.*

/**
 * Created by edwin on 07/11/2017.
 */

class EventTab : Fragment(), ViewPager.OnPageChangeListener {
    private var rootView: View? = null
    private var toolbar: Toolbar? = null
    private var tabLayout: TabLayout? = null
    private var viewPager: ViewPager? = null
    private val item: MenuItem? = null
    private var loading: LinearLayout? = null
    private var session: SessionManager? = null
    private var appBarLayout: AppBarLayout? = null
    private var myContext: FragmentActivity? = null
    private var adapter: ViewPagerAdapter? = null

    override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        rootView = inflater!!.inflate(R.layout.event_tab, container, false)
        initSession()
        renderView(rootView)
        setupViewPager()
        return rootView
    }

    private fun initSession() {
        session = SessionManager(activity)
    }

    private fun renderView(rootView: View?) {
        toolbar = rootView!!.findViewById<View>(R.id.tool_bar) as Toolbar
        loading = rootView.findViewById<View>(R.id.layout_loading) as LinearLayout
        appBarLayout = rootView.findViewById<View>(R.id.app_bar) as AppBarLayout
        appBarLayout!!.visibility = View.VISIBLE
        loading!!.visibility = View.GONE
        initToolbar()
    }

    fun setupViewPager() {
        viewPager = rootView!!.findViewById<View>(R.id.viewpager) as ViewPager
        adapter = ViewPagerAdapter(childFragmentManager)
        adapter!!.addFragment(EventListPromoFragment(), resources.getString(R.string.promo))
        adapter!!.addFragment(EventListNewFragment(), resources.getString(R.string.event_new))
        adapter!!.addFragment(EventListOtherFragment(), resources.getString(R.string.event_other))
        viewPager!!.adapter = adapter
        viewPager!!.offscreenPageLimit = adapter!!.count

        tabLayout = rootView!!.findViewById<View>(R.id.tabs) as TabLayout
        tabLayout!!.setupWithViewPager(viewPager)
    }

    internal inner class ViewPagerAdapter(manager: FragmentManager) : FragmentPagerAdapter(manager) {
        private val mFragmentList = ArrayList<Fragment>()
        private val mFragmentTitleList = ArrayList<String>()

        override fun getItem(position: Int): android.support.v4.app.Fragment {
            return mFragmentList[position]
        }

        override fun getCount(): Int {
            return mFragmentList.size
        }

        fun addFragment(fragment: android.support.v4.app.Fragment, title: String) {
            mFragmentList.add(fragment)
            mFragmentTitleList.add(title)
        }

        override fun getPageTitle(position: Int): CharSequence {
            return mFragmentTitleList[position]
        }
    }

    private fun initToolbar() {
        if (toolbar != null) {
            (activity as AppCompatActivity).setSupportActionBar(toolbar)
        }
        val actionBar = (activity as AppCompatActivity).supportActionBar
        if (actionBar != null) {
            (activity as AppCompatActivity).supportActionBar!!.title = resources.getString(R.string.news)
        }
    }

    override fun onPageScrolled(position: Int, positionOffset: Float, positionOffsetPixels: Int) {
        if (position == 0) {
            item!!.isVisible = false
        }
    }

    override fun onPageSelected(position: Int) {
        if (position == 0) {
            item!!.isVisible = false
        }
    }

    override fun onPageScrollStateChanged(state: Int) {}

    override fun onAttach(activity: Activity?) {
        myContext = activity as FragmentActivity?
        super.onAttach(activity)
    }

    override fun onResume() {
        super.onResume()

    }

    override fun onPause() {
        super.onPause()
    }
}